apps/system/js/browser_context_menu.js

```diff
@@ -250,12 +249,7 @@
   BrowserContextMenu.prototype.newWindow = function(manifest, isPrivate) {
     // For private windows we create an empty private app window.
     if (isPrivate) {
-      var privateBrowserUrl = location.origin + '/private_browser.html';
-      var config = new BrowserConfigHelper({url: privateBrowserUrl});
-      config.oop = true;
-      config.isPrivate = true;
-      var newApp = new AppWindow(config);
-      newApp.requestOpen();
+      browser.newPrivateWindow();
```

Comment:
Yup, that sounds better. Switched to using an event for now, will talk to you tomorrow about using service, and when it makes sense to use either. (I'm not currently clear on that)